### DISCUSSION TITLE
Support for DOOM?

### DISCUSSION THREAD

**Discussion by yliceee** [emacs-twist/twist.nix#142] [Q&A] [answered]
Hi, I'm super excited about this project. I have immediately started using and gained value from `magit-nix3`. I use DOOM emacs as a daily driver, and being able to manage my package repositories and versions in a flake would be amazing. In particular, atomic upgrades to packages without the equivalent of a time-consuming `doom sync -u` sounds awesome.

Without having taken too close of a look at twist.nix or twist.el, how difficult would it be to set up some integration between DOOM and emacs-twist?

**Comment by akirak** [comment, marked as answer]
Hello @rgri, thank you for your kind words.

> how difficult would it be to set up some integration between DOOM and emacs-twist?

I only tried out Doom a couple of years ago, and I don't have much experience with it. At a glance, Doom's `packages.el` files pin each package, which I don't think is easy to support with twist. 

If you ignore those pins, you will have to:

- Set up an initial boilerplate
- Ensure all package names you use are scanned by Nix (@LuigiPiucco has contributed https://github.com/emacs-twist/twist.nix/pull/60 for this)
- If your packages contain upstream bug (invalid recipe, missing dependencies, etc.), work around them
- If your packages have native dependencies, add build dependencies and overrides (usually `preBuild`)

The latter two steps are most time consuming, so it quite depends on how many packages you have and how many of them have quality issues or doing hard things. 

> atomic upgrades to packages without the equivalent of a time-consuming doom sync -u sounds awesome

I don't know how this is slow, but if you update a package in twist, all packages depending on it are also rebuilt. It may not be faster than Doom's updating system.

If you are still interested, the following resources may be helpful:

- @terlar's [emacs-config](https://github.com/terlar/emacs-config), actively maintained
- [An official repository for examples](https://github.com/emacs-twist/examples), unmaintained. Supporting [scimax](https://github.com/jkitchin/scimax) was rather hard.

Of course, I would be willing to help you with setting up twist, so please feel free to ask further questions.
